cmd/telemetrygen/internal/traces/worker.go

```diff
@@ -69,14 +78,15 @@ func (w worker) simulateTraces() {
 		_, child := tracer.Start(childCtx, "okey-dokey", trace.WithAttributes(
 			attribute.String("span.kind", "server"),
 			semconv.NetPeerIPKey.String(fakeIP),
-			semconv.PeerServiceKey.String("telemetrygen-client"),
+			semconv.PeerServiceKey.String(w.serviceName),
 		))
 
 		if err := limiter.Wait(context.Background()); err != nil {
 			w.logger.Fatal("limiter waited failed, retry", zap.Error(err))
 		}
 
 		opt := trace.WithTimestamp(time.Now().Add(fakeSpanDuration))
+		time.Sleep(time.Duration(rand.Intn(40)) * time.Millisecond)
```

Comment:
ditto, I don't get this, why is this needed? I don't think we should sleep if we can avoid it

Reply:
This seems like a bug somewhere else

Reply:
> This seems like a bug somewhere else

Agreed. I am okay with the other change (or something like it), but I don't think we should fix this with the `time.Sleep` approach, let's keep the PR only for the `loadSize` option and open an issue to discuss this bug and how to solve it.

Reply:
Just opened an issue here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22827